File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/player/Weapon.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player;

import java.io.Serializable;
import java.util.Random;

public enum Weapon implements Serializable {
    //these are fairly arbitrary
    //so we'll have to balance them later
    PULSE_LASER(10),
    BEAM_LASER(12),
    MILITARY_LASER(20);

    private int strength;

    /**
     * Creates a weapon with a given strength.
     *
     * @param strength how much damage the weapon does
     */
    Weapon(int strength) {
        this.strength = strength;
    }

    /**
     * @return the weapon's strength
     */
    public int getStrength() {
        return strength;
    }

    /**
     * @return a random weapon
     */
    public static Weapon getRandomWeapon() {
        //can do different weights later

        Random rand = new Random();
        return values()[rand.nextInt(values().length)];
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/player/Weapon.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

struct Weapon {
    public static let PULSE_LASER = Weapon(10)
    public static let BEAM_LASER = Weapon(12)
    public static let MILITARY_LASER = Weapon(20)
    
    var strength : Int
    
    public static let values = [
        PULSE_LASER,
        BEAM_LASER,
        MILITARY_LASER
    ]
    
    init(_ strength : Int) {
        self.strength = strength
    }
    
    static func getRandomWeapon() -> Weapon {
        return values[0]
    }
}
